homeassistant/components/holiday/config_flow.py

```diff
@@ -0,0 +1,89 @@
+"""Config flow for Holiday integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from babel import Locale
+from holidays import list_supported_countries
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_COUNTRY
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    CountrySelector,
+    CountrySelectorConfig,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import CONF_PROVINCE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_COUNTRIES = list_supported_countries(include_aliases=False)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_COUNTRY): CountrySelector(
+            CountrySelectorConfig(
+                countries=list(SUPPORTED_COUNTRIES),
+            )
+        ),
+    }
+)
+
+
+class HolidayConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Holiday."""
+
+    VERSION = 1
+
+    data: dict[str, Any] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self.data = user_input
+
+            selected_country = self.data[CONF_COUNTRY]
+
+            if SUPPORTED_COUNTRIES[selected_country]:
+                return await self.async_step_province()
+
+            locale = Locale(self.hass.config.language)
+            title = locale.territories[selected_country]
+            return self.async_create_entry(title=title, data=self.data)
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_province(self, user_input=None) -> FlowResult:
+        """Handle the province step."""
+        if user_input is not None:
+            combined_input: dict[str, Any] = {**self.data, **user_input}
+
+            locale = Locale(self.hass.config.language)
+            name = f"{locale.territories[combined_input[CONF_COUNTRY]]}, {combined_input[CONF_PROVINCE]}"
+
+            return self.async_create_entry(title=name, data=combined_input)
+
+        province_schema = vol.Schema(
+            {
+                vol.Optional(CONF_PROVINCE): SelectSelector(
+                    SelectSelectorConfig(
+                        options=SUPPORTED_COUNTRIES[self.data[CONF_COUNTRY]],
+                        mode=SelectSelectorMode.DROPDOWN,
+                        translation_key=CONF_PROVINCE,
+                    )
+                ),
+            }
+        )
```

Comment:
This seems to me something that belongs in the config entry setup flow, and not in the options flow.

Reply:
You mean like `STEP_USER_DATA_SCHEMA`? How can I pass `self.data[CONF_COUNTRY]` then?

Reply:
> How can I pass self.data[CONF_COUNTRY] then?

Make it a two step flow.

Reply:
Sorry, I still don't understand. Can you point me to some example or documentation?

Reply:
Just wanted to say I'm also not sure what the gap is here. I currently read the code as being a two step config flow and not part of the options flow. 

My impression is options flow means the flow where you change configuration options for the integration after it is set up, which I don't see being used here.

Probably something subtle I'm missing so clarification would be appreciated, thanks!

Reply:
It is a two-step flow so I assume frenck misread the code